projects/telemetry/inc/nmea.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+// This module is used to parse and represent NMEA data. Used in the GPS module
+
+#include <stdint.h>
+#include <stdio.h>
+#include "status.h"
+
+// This enum contains the list of NMEA sentences (not all of then are supported yet)
+// because they do not contain useful information
+typedef enum {
+  NMEA_MESSAGE_ID_UNKNOWN = 0,
+  NMEA_MESSAGE_ID_GGA,
+  NMEA_MESSAGE_ID_GLL,
+  NMEA_MESSAGE_ID_GSA,
+  NMEA_MESSAGE_ID_GSV,
+  NMEA_MESSAGE_ID_RMC,
+  NMEA_MESSAGE_ID_VTG,
+  NUM_NMEA_MESSAGE_ID_TYPES
+} NmeaMessageId;
+
+// An enum to represent a position fix's validity (defined by 1, 2, or 6 as described in
+// https://www.linxtechnologies.com/wp/wp-content/uploads/rxm-gps-f4.pdf
+// on page 14)
+
+typedef enum {
+  NMEA_PF_INVALID = 0,
+  NMEA_PF_GPS_SPS = 1,
+  NMEA_PF_DIFFERENTIAL_GPS_SPS = 2,
+  NMEA_PF_DEAD_RECKONING_MODE = 6,
+  NUM_NMEA_POSITION_FIX_TYPES = 4
+} NmeaPositionFix;
+
+typedef struct {
+  uint8_t hh;    // Hours
+  uint8_t mm;    // Minutes
+  uint8_t ss;    // Seconds
+  uint16_t sss;  // Milliseconds
+} NmeaUtcTime;
+
+// Representation of longitude or latitude
+// https://en.wikipedia.org/wiki/Longitude
+typedef struct {
+  uint16_t degrees;
+  uint16_t minutes;
+  uint16_t fraction;
+} NmeaCoord;
+
+// Info passed from the GPS chip should be dropped into this struct (more fields
+// coming soon)
+
+typedef struct {
+  NmeaPositionFix position_fix;  // True if this struct has valid data
+  uint16_t satellites_used;
+  uint16_t adc;                        // Age of diff. corr. in seconds
+  uint16_t drs;                        // Diff. Ref. Station. Not sure what it is yet
+  uint16_t hdop_integer;               // Horizontal dilution of precision, characteristic
+  uint16_t hdop_fraction;              // Horizontal dilution of precision, mantissa
+  uint16_t msl_altitude_integer;       // In meters, characteristic
+  uint16_t msl_altitude_fraction;      // In meters, mantissa
```

Comment:
What unit are fractions in?

Reply:
Looks like you found a bug. It would parse `02` and `2` as `2` which is not correct. Fixing now

Reply:
But the actual unit will be a thousandth of the integer unit

Reply:
Okay, can you add a test for that and add a comment for the unit? If that holds true for all fractions in this module, you can use a module-level comment.

Reply:
Ok, I will do this later today